src/planscape/collaboration/utils.py

```diff
@@ -0,0 +1,118 @@
+from collaboration.models import Collaborator, Permissions, Role
+from planning.models import PlanningArea, Scenario
+from django.contrib.auth import get_user_model
+from django.contrib.contenttypes.models import ContentType
+
+
+User = get_user_model()
+
+
+def is_creator(user: User, planning_area: PlanningArea):
```

Comment:
One thing that concerns me about these functions is that the `is_creator()` and  `get_collaborator()` functions are designed exclusively for `planning_area` models. 

 It seems inevitable that we'll use object permissions like this for other types of objects in the future, so maybe it's worth renaming them or refactoring where we provide them.

I don't know if they deserve to be encapsulated in a class like PlanningAreaPermissions or if they should just be renamed for clarity like `get_planningarea_roles()`, so we know they're hardcoded for planning area records.

However, since we're already looking up the `content_type` for every planning area in `get_collaborator`, we could also rewrite that so it is a much more generic function that accepts any object like `given_object`, then checks for the `content_type_id` using something like `ContentType.objects.get_for_model(given_object).id` 
Let me know if that makes sense. I think this is hard to explain without specific examples;

Reply:
thanks! Yeah i think i get it, something like `is_creator(user:User, content_type: ContentType, id)`. Good call!

Reply:
I like that idea! but Im having trouble on how to implement it, because i'll be checking user_id  == given_object.user.id ... so its something like `ContentType.objects.get_for_model(given_object).user.pk == user.pk` ?
If that's the case, why do I need the content_type... just do `given_object.user.pk`